### ISSUE TITLE
Training a model from scratch, not fine-tuning.

### ISSUE THREAD

**Issue by MatthewWaller** [huggingface/diffusers#1181]
**Is your feature request related to a problem? Please describe.**
For philosophical/ethical reasons, I would like to try my hand at create a version of stable diffusion that uses only public domain images.

**Describe the solution you'd like**
I would like an example in the training scripts that show how to get a version of Stable Diffusion started training from scratch.

**Describe alternatives you've considered**
I've looked into the fine-tuning scripts, but I want to make sure that I'm aware of any best practices when it comes to initializing and creating an entire model.

**Additional context**
I can worry about the cost of training and finding the data, of course, but even having the basic knowledge of getting everything started from scratch would be a boon!

**Comment by patrickvonplaten** [comment]
Hey @MatthewWaller, 

Yes at the moment, it might maybe be best to have a look at the CompVis repo. We don't yet have a script to train stable diffusion from scratch

**Comment by ddxue** [comment]
Just curious about this too, would the idea here that we would need to initialize the `unet` weights instead of loading from `diffusion_pytorch_model.bin`? 

Is there some hack-y approach we could take to initialize the `unet` weights as if training from scratch instead of load them?

**Comment by patrickvonplaten** [comment]
Hey @ddxue,

Exactly you can / should use the `from_config(...)` method for this. In short the following should work.

```python
from diffusers import UNet2DConditionModel

config = UNet2DConditionModel.load_config("CompVis/stable-diffusion-v1-4", subfolder="unet")
random_init_model = UNet2DConditionModel.from_config(config)
```

**Comment by MatthewWaller** [comment]
Sorry to keep commenting, but @patrickvonplaten if I were to load the vae and unet from config so that they’re untrained, could the fine tuning script be use to train them from scratch then? Or are there more steps I would need to do?

**Comment by patrickvonplaten** [comment]
Hey @MatthewWaller,

Yes technically you could do this. However, the VAE should probably be trained separately just like it was done for Stable Diffusion since the VAE should be able to encoder & decode images independently of the text prompt. 

The UNet should be trainable just by using the fine-tuning script from scratch.

**Comment by MatthewWaller** [comment]
Hey thanks @patrickvonplaten ! For the VAE, it accepts the output of the Unet though, correct? Or do I need to just apply noise to images with the scheduler to train? In other words, any guidance y’all could give about how to train the vae? I looked through CompVis but nothing stood out.

**Comment by patrickvonplaten** [comment]
In general, it is not recommended to train the VAE at all. CompVis didn't train the VAE for Stable Diffusion

**Comment by mtngld** [comment]
> In general, it is not recommended to train the VAE at all. CompVis didn't train the VAE for Stable Diffusion

Do you know which VAE weights did they use? The model trained in https://github.com/CompVis/latent-diffusion?

**Comment by MatthewWaller** [comment]
Ah, gotcha @patrickvonplaten . So maybe I can keep that frozen. However, will it matter if I am using a different image size? Like if I wanted to train a tiny stable diffusion from CIFAR10, does it make sense to use the pre-tained VAE on 32x32 images? Probably not, I'm guessing.

**Comment by patrickvonplaten** [comment]
Good question! Actually 32x32 is smaller than the latent space of the diffusion model (which is 64x64) so in this case using a latent diffusion model is probably overkill. 

Two pointers that could help in this case:
 - Maybe you find some smaller more adapted VAE models here: https://github.com/CompVis/taming-transformers#overview-of-pretrained-models
 - (probs better) Don't use a VAE at all and rather use a model like UNClip: https://github.com/huggingface/diffusers/blob/b15a951a48016a96b7db24090ab3f0c824ad4e73/src/diffusers/pipelines/unclip/pipeline_unclip.py#L35 or DiT instead: https://github.com/huggingface/diffusers/blob/b15a951a48016a96b7db24090ab3f0c824ad4e73/src/diffusers/pipelines/dit/pipeline_dit.py#L31

**Comment by jfischoff** [comment]
I just want to add that having a low resolution stable diffusion model for prototyping would be very helpful for those hacking on pipeline extensions and trying to train new ControlNets. I realize it is complicated unfortunately.

**Comment by ana-mariacretu** [comment]
Hello, the most recent version of train_text_to_image.py supports EMA training. Should the ema_unet be initialized as a deep copy of the unet (which is initialized from scratch as described previously)? https://github.com/huggingface/diffusers/blob/main/examples/text_to_image/train_text_to_image.py#L626
